experiments/utils.py

```diff
@@ -342,21 +348,44 @@ def _session_enrollment_latest_version(data):
 
 class SessionUser(WebUser):
     def __init__(self, session, request=None):
+        self._enrollment_cache = {}
         self.session = session
         self.request = request
+        self.goals_key = SESSION_USER_GOALS_REDIS_KEY % self._participant_identifier()
         super(SessionUser, self).__init__()
 
     def _get_enrollment(self, experiment):
-        enrollments = self.session.get('experiments_enrollments', None)
-        if enrollments and experiment.name in enrollments:
-            alternative, _, _, _ = _session_enrollment_latest_version(enrollments[experiment.name])
-            return alternative
-        return None
+        if experiment.name not in self._enrollment_cache:
+            try:
+                self._enrollment_cache[experiment.name] = SessionEnrollment.objects.get(session_key=self._session_key, experiment=experiment).alternative
+            except SessionEnrollment.DoesNotExist:
+                self._enrollment_cache[experiment.name] = None
```

Comment:
where not fetch all the experiments for a user and cache them here? Not thought these through, but might be more effecient